src/charm.py

```diff
@@ -184,73 +151,72 @@ def reconfigure(self, event):
-        if not self.cluster.replica_set_initialized:
-            self.unit.status = WaitingStatus("Initializing the replica set")
-            self.mongo.initialize_replica_set(self.cluster.hosts)
-            self.on.replica_set_configured.emit(self.cluster.hosts)
-
-        self.on.cluster_ready.emit()
-        self.on_update_status(event)
-        logger.debug("Running on_mongodb_started finished")
+    def on_database_relation_changed(self, event):
+        event.relation.data[self.unit]['replicated'] = str(self.is_joined)
+        event.relation.data[self.unit][
+            'replica_set_name'] = self.replica_set_name
+        event.relation.data[self.unit]['standalone_uri'] = "{}".format(
+            self.standalone_uri)
+        event.relation.data[self.unit]['replica_set_uri'] = "{}".format(
+            self.replica_set_uri)
 
     ##############################################
-    ############### PROPERTIES ###################
+    #               PROPERTIES                   #
     ##############################################
 
     @property
     def mongo(self):
         return Mongo(
-            standalone_uri=self.cluster.standalone_uri,
-            replica_set_uri=f"{self.cluster.replica_set_uri}?replicaSet={self.replica_set_name}",
+            standalone_uri=self.standalone_uri,
+            replica_set_uri=f"{self.replica_set_uri}?replicaSet={self.replica_set_name}",
         )
 
+    @property
+    def replica_set_uri(self):
+        uri = "mongodb://"
+        for i, host in enumerate(self.cluster_hosts):
+            if i:
+                uri += ","
+            uri += f"{host}:{self.port}"
+        uri += "/"
+        return uri
+
+    @property
+    def standalone_uri(self):
+        return f"mongodb://{self.model.app.name}:{self.port}/"
+
     @property
     def replica_set_name(self):
         return self.model.config["replica_set_name"]
 
     @property
-    def standalone(self):
-        return self.model.config["standalone"]
+    def num_peers(self):
+        return len(self.peer_relation.units) + 1 if self.is_joined else 1
 
-    ##############################################
-    ############## PRIVATE METHODS ###############
-    ##############################################
-
-    def _check_settings(self):
-        problems = []
-        config = self.model.config
+    @property
+    def is_joined(self):
+        return self.peer_relation is not None
 
-        for setting in REQUIRED_SETTINGS:
-            if config.get(setting) is None:
-                problem = f"missing config {setting}"
-                problems.append(problem)
-        if not self.standalone:
-            for setting in REQUIRED_SETTINGS_NOT_STANDALONE:
-                if not config.get(setting):
-                    problem = f"missing config {setting}"
-                    problems.append(problem)
+    def _get_unit_hostname(self, _id: int) -> str:
+        return f"{self.model.app.name}-{_id}.{self.model.app.name}-endpoints"
```

Comment:
This line is not Python 3.5 compatible. 
What about this:

```python
"{self.model.app.name}-{_id}.{self.model.app.name}-endpoints".format(self.model.app.name, _id, self.model.app.name)
```

Reply:
Thanks again. I feel silly now for not having paid attention to these details.